src/test/java/org/dz/solution/task31/Task31Test.java

```diff
@@ -0,0 +1,74 @@
+package org.dz.solution.task31;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class Task31Test {
+
+    public static final String WORD_DELIMITER = " ";
+
+    @Test
+    public void writeReadStringMatrixFile() throws IOException {
+        String[][] expectedMatrix = {
+                {"7", "Ок", "Коля", "13", "утка", "45"},
+                {"Phone", "2,5", "string", "@", null, null},
+                {"собака", null, null, null, null, null},
+                {"%", "81", "Задание", "cat", "слово", null}
+        };
+        String filePath = "src/main/resources/org/dz/solution/task31/StringMatrix";
+        Files.delete(Path.of("src/main/resources/org/dz/solution/task31/StringMatrix"));
+
+        String filePathTwo = "src/main/resources/org/dz/solution/task31/StringMatrix";
+        PrintWriter pw = new PrintWriter(filePathTwo);
```

Comment:
The file must be closed!

Reply:
Wrap the loop to a method :
PrintWriter pw = new PrintWriter(filePathTwo);
        for (String[] matrix : expectedMatrix) {
            for (String s : matrix) {
                if (s == null) {
                    break;
                }
                pw.print(s + " ");
            }
            pw.println();
        }
        pw.close();